### PR TITLE
Kakute H7 v2 and Kakute H7 mini v2

### PR DESCRIPTION
**Pull request by andyp1per**
Support the new Holybro Kakute H7 v2 and Kakute H7 Mini v2.

### CODE REVIEW

File: libraries/AP_Motors/AP_MotorsMatrix.cpp [hunk 1]
```diff
@@ -1057,6 +1057,24 @@ void AP_MotorsMatrix::setup_motors(motor_frame_class frame_class, motor_frame_ty
...
+                    _frame_type_string = "BF_X";
+                    static const AP_MotorsMatrix::MotorDef motors[] {
+                        {  135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  2 },
+                        {   45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 1 },
+                        { -135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 3 },
+                        {  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  4 },
+                        {  135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 6 },
+                        {   45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  5 },
+                        { -135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  7 },
+                        {  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 8 },
```
Comment (IamPete1) [inline libraries/AP_Motors/AP_MotorsMatrix.cpp:1073]:
```suggestion
                        {  135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  3 },
                        {   45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 1 },
                        { -135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 5 },
                        {  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  7 },
                        {  135, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 4 },
                        {   45, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  2 },
                        { -135, AP_MOTORS_MATRIX_YAW_FACTOR_CW,  6 },
                        {  -45, AP_MOTORS_MATRIX_YAW_FACTOR_CCW, 8 },
```
I think the motor test order should be this. This is the diagram were trying to match:
image

**Review by Hwurzburg — COMMENTED** [review]
-cant use FRSKY on T6 and half duplex RC protocols on R6 as implied
-why is a new motors config in this PR? shouldn't be a separate PR?
- are all the sensor changes only used by these boards? again, shouldn't they be separate?

**Comment by andyp1per** [comment]
@Hwurzburg this is a new board that supports two 4-in-1 ESCs, the PWM change make RPM telemetry work on all of them. I originally did the BMI270 driver but have never flown it, now that it is becoming more common I did a thorough review and found some issues. TX6/RX6 is a regular serial UART as per the Holybro docs so can be used for anything and in particular so that it can be used as the Holybro docs indicate . PPM is not supported but neither is it documented on the Holybro site and neither it is supported by BF. If we want to do a PPM version we can, but I don't think its worth it. Can do separate PR's but its already been reviewed and the driver commits are not entangled with anything else so they would show up in the codebase in the same way.

**Comment by andyp1per** [comment]
Flew on my 5" beater - IMU looks comparable to ICM42688, so all good.

**Review by tridge — APPROVED** [review]

**Comment by andyp1per** [comment]
@rmackay9 / @tridge I think this should go into 4.2

**Pull request merged**